### ISSUE TITLE
Golang intellisense stopped working after upgrading to Go v1.13 when using gopls

### ISSUE THREAD

**Issue by saurabh-hirani** [microsoft/vscode-go#2837]
I have encountered this issue often on VSCode upgrade - autocompletions of imported and unimported packages stop working after updating go version but either a reload window or stopping/starting ```gocode ``` fixes it. But this time I tried all of the following but it didn't work:

1. Upgraded go to v1.13 - updated GOROOT accordingly.
2. Did "Go: Install / Update tools" with ``` gocode ``` stopped and started it after the update.
3. Reloaded, stopped, started, uninstalled, reinstalled VSCode.
4. Stopped, started gopls process. 

None of the above helped. I also checked my user settings to see if any setting has changed. But I still have the following settings enabled:

```
    "go.useLanguageServer": true,
    "go.autocompleteUnimportedPackages": true,
```

I also removed my old settings and reinstalled VSCode by following - https://medium.com/@jimkang/complete-uninstall-remove-vscode-mac-5e48bef3bdec but it didn't resolve the issue.

I raised the query in https://gitter.im/Microsoft/vscode-go?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge but didn't get a response - hence raising the same here.

**Comment by stamblerre** [comment]
The autocompletion of unimported packages is being moved into `gopls`, so I think the more recent versions of the extension have disabled the additional unimported completion items provided by the extension (they did not work well with `gopls`'s ranking). You can try out the feature through `gopls` by adding the following to your settings:

```json5
"gopls": {
    "completeUnimported": true
}
```
It currently only works for the standard library, but packages in the module cache will be supported soon.

**Comment by saurabh-hirani** [comment]
Thanks @stamblerre  - I did add that but there wasn't any impact and the option seemed to be grayed out. 

![Screenshot 2019-10-17 at 2 39 18 PM](https://user-images.githubusercontent.com/485317/66983948-53c6db00-f0ec-11e9-9cb2-11c4627ba990.png)

doing a ```fmt.``` does not show up auto completion and I have to import the fmt package to get the options. That wasn't the case earlier. For stdlib it used to work out of the box.

**Comment by stamblerre** [comment]
The grayed out setting is expected, while `gopls` configurations are still in flux, we haven't added them to the VSCode extension yet. Make sure to reload the language server after you change the settings, as it needs to be restarted to reflect the settings. After that it, it should suggest `fmt` to you.

**Comment by saurabh-hirani** [comment]
Hi @stamblerre - I did all of that and yet ```fmt.``` does not autocomplete and shows the following output in the language server

<img width="565" alt="Screenshot 2019-10-18 at 8 17 32 AM" src="https://user-images.githubusercontent.com/485317/67056771-17dd5580-f180-11e9-8892-939c5e9b8d03.png">


I guess I will stick with doing a ```fmt.Println``` to get the initial import working and then use autocompletion later. And for third party modules installed via go get - I will have to do an explicit import. Hoping to see this become a seamless user experience in the future. I understand that supporting so many platforms and versions is hard and I appreciate the work your team is putting in for making vscode-go so awesome!

**Comment by stamblerre** [comment]
What version of `gopls` are you using (`gopls version`)? If the version is not `v0.1.7` then that may be the issue, but you should definitely be able to get autocompletion of unimported standard library packages.

Also, you should use the suggested settings [here](https://github.com/golang/tools/blob/master/gopls/doc/vscode.md) to get import organization to happen automatically for third-party packages.

**Comment by saurabh-hirani** [comment]
@stamblerre  I am using the latest version

```
gopls version
golang.org/x/tools/gopls v0.1.7
    golang.org/x/tools/gopls@v0.1.7 h1:YwKf8t9h69++qCtVmc2q6fVuetFXmmu9LKoPMYLZid4=
```

I tried those settings, reloaded the window, restarted the language server and yet there is no change in the behaviour. But doing the first ```fmt.Println()``` imports ```fmt``` and autocompletion turns on after that.

**Comment by stamblerre** [comment]
When you are triggering the completions for `fmt`, are you typing `fmt.` and then triggering completions? If so, they won't work as you expect because the import hasn't been added yet. However, if you begin typing `fmt`, the completion item for `fmt` should pop up. If you select it, the import will be added, and you should then be able to get those completions shown to you.

I understand this experience isn't very intuitive, but we are working on improving it.

**Comment by alecthomas** [comment]
I have the same problem, however I don't get a full completion list when I just type "fmt":

![image](https://user-images.githubusercontent.com/41767/67169740-f89d2d00-f3f8-11e9-97b1-92c280536567.png)

If I then type `fmt.<ctrl-space>` I get:

![image](https://user-images.githubusercontent.com/41767/67169746-09e63980-f3f9-11e9-93a8-0ec688187e7b.png)

If I then append a `fmt.P<ctrl-space>` (which I know exists), it finally works:

![image](https://user-images.githubusercontent.com/41767/67169768-38641480-f3f9-11e9-9e54-091c404d2bbe.png)

So, basically, I need to know what I'm looking for in order for completion to work...

This is on Go 1.13.3 and the same version of gopls as @saurabh-hirani is on.

For reference this is after [this line](https://github.com/alecthomas/participle/blob/master/parser.go#L28).

**Comment by saurabh-hirani** [comment]
@stamblerre thanks - tried it out - it works. I was doing ```fmt.``` and waiting for the completion. Doing ```fmt```, selecting the module from the auto-complete drop down and the doing ```fmt.``` gives the answer. Thanks for being so patient and guiding me through. Will use the prescribed way of importing for now.

**Comment by saurabh-hirani** [comment]
@alecthomas I did the following and it works for me:

1. Checked gopls version as per above..
2. Just for good measure - reloaded the window and restarted the gopls server.
3. Wrote ```fmt``` and then selected the ```{} fmt``` entry from the dropdown.
4. That added the line ```import "fmt"``` at top of my code.
5. Wrote ```fmt.``` and the auto complete options popped up.

Please not this works only for stdlib. For 3rd party you have to write upto name of the function to get completion e.g. for the https://github.com/subosito/gotenv package I have to type ```gotenv.Load``` and then  only does it add ```import "github.com/subosito/gotenv"``` at the top. 

I tried with ```parser``` and am able to get autocomplete as follows:

<img width="804" alt="Screenshot 2019-10-21 at 8 58 51 AM" src="https://user-images.githubusercontent.com/485317/67169945-47d76380-f3e1-11e9-8668-d5f18f978acd.png">

**Comment by alecthomas** [comment]
@saurabh-hirani that's not equivalent. `parser` in my example is a local variable, not a package.

**Comment by alecthomas** [comment]
(that said, maybe these are different issues)

**Comment by hgedia** [comment]
Chipping in here , with gopls 0.1.7/go 1.13 , on hovering the function , the tool tip stopped appearing and could not jump to function definitions. Downgraded for now to go 1.12 and its back up.  Happy to try out a few things if needed.

**Comment by stamblerre** [comment]
@alecthomas and @hgedia, it does sound like you are experiencing two separate issues. Do you minding filing them on the [Go issue tracker](https://github.com/golang/go/issues/new), along with the information requested in the [Troubleshooting guide](https://github.com/golang/tools/blob/master/gopls/doc/troubleshooting.md)?

**Comment by stamblerre** [comment]
A lot has changed in `gopls` since this issue was filed, so @ramya-rao-a, I think that this can be closed. 
Please report new issues if you experience problems with the current version of `gopls`.

**Comment by alecthomas** [comment]
FWIW I tried VSCode again with gopls and it seems marginally better, in that it worked for an hour or two before failing again. I'll file another issue.

**Comment by alecthomas** [comment]
Filed #3005. This is the same issue I was encountering in October, so nothing has changed unfortunately :(

**Comment by ramya-rao-a** [comment]
Thanks @alecthomas 

We will follow up in the new issue

**Comment by amirhj** [comment]
I removed the .vscode folder in my linux machine and started the vscode again and installed every extension again and it is working now!
